packages/core/src/utils-hoist/instrument/fetch.ts

```diff
@@ -48,6 +48,16 @@ function instrumentFetch(onFetchResolved?: (response: Response) => void, skipNat
 
   fill(GLOBAL_OBJ, 'fetch', function (originalFetch: () => void): () => void {
     return function (...args: any[]): void {
+      // We capture the stack right here and not in the Promise error callback because Safari (and probably other
+      // browsers too) will wipe the stack trace up to this point, only leaving us with this file which is useless.
+
+      // NOTE: If you are a Sentry user, and you are seeing this stack frame,
+      //       it means the error, that was caused by your fetch call did not
+      //       have a stack trace, so the SDK backfilled the stack trace so
+      //       you can see which fetch call failed.
+      const virtualError = new Error();
+      const virtualStackTrace = virtualError.stack;
```

Comment:
It is relatively expensive for engines to compute stack traces. I wonder if we could change this a bit so we don't create stack traces every time fetch is called but only when we actually need it.

Reply:
Ok, passing a boolean whether the fetch/xhr instrumentations are triggered by HttpClient was flaky because of `maybeInstrument`, so passing the `error` itself in handler data under `virtualError` key (not to conflict with other integrations like Breadcrumbs that use `error` key) and creating the stacks inside HttpClient will hopefully give us a slight performance improvement. https://github.com/getsentry/sentry-javascript/pull/14515/commits/a7b89bce01397d3babbcaceec7ee9a8f051420a2